### DISCUSSION TITLE
Help wanted

### DISCUSSION THREAD

**Discussion by ransome1** [ransome1/sleek#388] [Ideas]
`Amazing feature requests, major and minor bug reports are piling up, and great discussions are ongoing. I want to thank you for supporting this project. But I also need to admit that I'm overwhelmed by the sheer amount of work. Please understand that sleek is FOSS and I'm not aiming to earn money with it, so my time on the project is limited. If you are skilled in HTML, CSS, JavaScript, the Electron framework, writing Playwright test cases, improving UX/UI or if you're able to maintain sleek for one of the distribution channels, please check in with me https://github.com/ransome1/sleek/discussions/388 or check the [issue list](https://github.com/ransome1/sleek/issues) and start working on bugs and feature requests.`

Let's use this thread to get in touch with each other and align on what we can work on together. There is plenty stuff to do, like
- Bug fixing
- Code refactoring
- Adding features
- Writing missing test cases
- Improving UX/UI
- Adding or improving translations
- Managing the issues list on GitHub
- Improving automation of sleeks distribution

**Comment by puka-tchou** [comment]
I just stumbled upon this project, I'll try it to see if I keep using it. If so, expect some help from me quickly :)

Good luck with keeping your project ongoing and thank you for this great piece of software!

**Reply by ransome1** [reply]
Thanks for reaching out. Would be happy to have you on board.

**Comment by DianaAvilesCh** [comment]
Dear Robin 

Please find enclosed this letter requesting authorization to participate in the development of the End of Course Project "Incorporation of the Help Organization Technique according to Use Cases in the Open Source Software Development Process" in the Sleek project.
Diana Stefania Aviles Chalaco,
Lister Leonardo Mera Carbo,
Jaime Andrés Mora Vera.

[LETTER OF AUTHORIZATION.pdf](https://github.com/ransome1/sleek/files/8983749/LETTER.OF.AUTHORIZATION.pdf)

**Comment by ransome1** [comment]
Dear @DianaAvilesCh , Lister and Jaime. 

Thank you for getting in touch with the project. 

I've read the LOA and you can consider your request granted. I think the research you're proposing can benefit sleek significantly.

I understand that your research will result in feedback on usability, features and bugs. If you plan to publish these in the "Issues" section on Github, please make sure to check if there is already a corresponding issue report in order to avoid duplicates. 

The project is happy to have you participating. If you wan to get in touch with me directly please drop me an email at ransome@duck.com

Best
Robin

**Comment by xpanvictor** [comment]
Ill like to help with the project.
Ive cloned the repo and will start checking out the issues.

**Comment by CPUGPU** [comment]
Did the sudden popularity scare everybody away, including ransome1?  :(

**Comment by ransome1** [comment]
@CPUGPU No, don't worry. But I've been afk for personal reasons until recently and now back. Still looking for support.

@xpanvictor lovely, glad to hear. If you have questions, don't hesitate to get in touch with me.

**Comment by bennettscience** [comment]
I just forked a copy and I'm hoping to be able to tackle some of the UI requests (keybinding popups, etc). Do you have a particular format for PRs you'd like? Or just make sure tests pass?

**Reply by ransome1** [reply]
> I just forked a copy and I'm hoping to be able to tackle some of the UI requests (keybinding popups, etc). Do you have a particular format for PRs you'd like? Or just make sure tests pass?

Happy to hear :) I don't have a particular format for PRs. Of course it would be great if you could write playwright testcases for whatever you're adding or improving. I' seriously behind on writing cases and need to spend more time on that as well. Get yourself setup and check in with me, if you have questions.

**Comment by starseeker-code** [comment]
Hi, ransome1.

I'm a (senior) python developer and a (mid level) javascript developer, my frontend skills are a little bit shaky :disappointed: (specially with CSS :D) but I'm volunteering to help in this project. I love sleek, and I'd love to help if you need me.

Please, could you provide me any contributions guidelines? How should I structure my forks/branches/PRs? Have you considered Typescript? Do you have any discord/slack?

Warm regards :smile:

**Comment by Jinu-Vijayan** [comment]
Hello , ransome1.

I would like to help with the project. Please provide me with any contribution guidelines you have for the project.

**Comment by ransome1** [comment]
Hey everyone! Thank you for reaching out, and I apologize for the delayed response, @starseeker-code.

I wanted to let you know that I've made the decision to rewrite sleek to make it more accessible for your participation. The rewrite will be done using React and will primarily utilize TypeScript.

The current progress of the rewrite can be found in the `2.0` branch. While the basic functionality is in place and it can be launched, it's still too early to use it or start adding new features.

I would like to invite you to check out the current development state and reach out to me if you're interested in helping with the rebuilding of sleek. We can discuss which part you would like to work on. In the upcoming push, I will also publish some contributing guidelines to provide more information.

Thank you again for your interest, and I look forward to collaborating with you on sleek's rewrite!
